hooks/QuizBuilder/useQuizBuilder.ts

```diff
@@ -0,0 +1,280 @@
+      (question) => question.id === questionId
+    );
+    if (questionToBeUpdated) {
+      setQuestions((prev) =>
+        prev.map((question) => {
+          if (question === questionToBeUpdated) {
+            const mcq = questionToBeUpdated as MultipleChoiceQuestion;
+            if (action === UPDATE_ACTIONS.ADD) {
+              return {
+                ...questionToBeUpdated,
+                options: [
+                  ...mcq.options,
+                  { id: uuid(), content: "", correct: false },
+                ],
+              } as MultipleChoiceQuestion;
+            } else if (action === UPDATE_ACTIONS.REMOVE && option) {
+              return {
+                ...questionToBeUpdated,
+                options: mcq.options.filter(
+                  (currOption) => currOption.id !== option.id
+                ),
+              } as MultipleChoiceQuestion;
+            } else if (action === UPDATE_ACTIONS.UPDATE && option) {
+              return {
+                ...questionToBeUpdated,
+                options: mcq.options.map((currOption) => {
+                  if (currOption.id === option.id) {
+                    return option;
+                  }
+                  // this gives the radio select behaviour
+                  // if one option is correct than all others
+                  // should be incorrect
+                  if (option.correct) {
+                    return { ...currOption, correct: false };
+                  }
+                  return currOption;
+                }),
+              } as MultipleChoiceQuestion;
+            }
+          }
+          return question;
+        })
+      );
+    }
+  };
+
+  const updateQuestionPoints = (questionId: string, points: number) => {
+    const questionToBeUpdated = questions.find(
+      (question) => question.id === questionId
+    );
+    if (questionToBeUpdated) {
+      setQuestions((questions) =>
+        questions.map((question) => {
+          if (question.id === questionId) {
+            return {
+              ...question,
+              points: isNaN(points) || points < 0 ? 0 : points,
+            };
+          }
+          return question;
+        })
+      );
+    }
+  };
+
+  const updateQuestionAnswerKey = (questionId: string, answerKey: string) => {
+    const questionToBeUpdated = questions.find(
+      (question) => question.id === questionId
+    );
+    if (questionToBeUpdated) {
+      setQuestions((questions) =>
+        questions.map((question) => {
+          if (question.id === questionId) {
+            return { ...question, answerKey: answerKey } as EssayQuestion;
+          }
+          return question;
+        })
+      );
+    }
+  };
+
+  const validateQuiz = () => {
+    const errorMap: QuizValidationError = new Map();
+    for (const question of questions) {
+      const currentQuestionErrors = [];
+      if (!question.prompt?.trim()) {
+        currentQuestionErrors.push("Missing question prompt");
```

Comment:
Hello Rashad! So apparently we're displaying the errors as a list, this can help our users to know what mistakes they have done immediately. Nice one. 😁

Reply:
Yes Alex, I'm actually planning to make it more granular in the future, with the errors being tied to each element rather than the whole question

Reply:
Alright shad. Thanks! 😁